master/internal/task/allocation.go

```diff
@@ -444,6 +445,8 @@ func (a *Allocation) TaskContainerStateChanged(
 			prom.RemoveAllocationReservation(a.reservations[cID].Summary())
 		}
 	}
+
+	_ = a.db.UpdateAllocationState(a.req.AllocationID, a.state, a.logBasedReadinessPassed)
```

Comment:
ditto, have to handle the error here, also i think it makes sense to make is_ready NULL in the db and just null it out when we're done (like we do with progress, since it's meaningless once completed)